From 44fa30cd63eb543d04a67adf66fa2502ab121d60 Mon Sep 17 00:00:00 2001
From: andy <66240551+flandolf@users.noreply.github.com>
Date: Sat, 29 Nov 2025 12:53:02 +1100
Subject: [PATCH] allow platform_app to set touch sensitivity property

Change-Id: I31dc31d2da2d987d0ffbcd3e6af487ef362c2f16

diff --git a/common/private/platform_app.te b/common/private/platform_app.te
index 585b032..610c919 100644
--- a/common/private/platform_app.te
+++ b/common/private/platform_app.te
@@ -10,3 +10,6 @@ hal_client_domain(platform_app, hal_lineage_powershare)
 get_prop(platform_app, usb_uvc_enabled_prop)
 
 get_prop(platform_app, bluetooth_lea_prop)
+
+# Allow SystemUI to set touch sensitivity prop
+allow platform_app debug_prop:property_service set;
\ No newline at end of file
-- 
2.52.0

