From 6b2e10255ea71909870492a0871423ff1e659fc9 Mon Sep 17 00:00:00 2001
From: flandolf <66240551+flandolf@users.noreply.github.com>
Date: Thu, 14 Aug 2025 19:15:50 +1000
Subject: [PATCH] Disable treble sepolicy tests

Change-Id: Ifff22e8a1425b15a28d324b08efe0ff29e9bebf1

diff --git a/Android.bp b/Android.bp
index 7d62f26b8..7297d0c2a 100644
--- a/Android.bp
+++ b/Android.bp
@@ -1264,12 +1264,6 @@ phony {
             "sepolicy_compat_test",
             "sepolicy_test",
             "sepolicy_dev_type_test",
-            "treble_sepolicy_tests_29.0",
-            "treble_sepolicy_tests_30.0",
-            "treble_sepolicy_tests_31.0",
-            "treble_sepolicy_tests_32.0",
-            "treble_sepolicy_tests_33.0",
-            "treble_sepolicy_tests_34.0",
         ],
     }) + select((
         soong_config_variable("ANDROID", "PLATFORM_SEPOLICY_VERSION"),
@@ -1279,7 +1273,6 @@ phony {
         ("202404", true, true): [],
         (default, true, true): [],
         (default, default, default): [
-            "treble_sepolicy_tests_202404",
         ],
     }) + select((
         soong_config_variable("ANDROID", "PLATFORM_SEPOLICY_VERSION"),
@@ -1290,7 +1283,6 @@ phony {
         ("202504", true, true): [],
         (default, true, true): [],
         (default, default, default): [
-            "treble_sepolicy_tests_202504",
         ],
     }) + select(soong_config_variable("ANDROID", "RELEASE_BOARD_API_LEVEL_FROZEN"), {
         true: ["se_freeze_test"],
-- 
2.48.1

